src/popover.ts

```diff
@@ -888,6 +888,20 @@ export class HoverEditor extends nosuper(HoverPopover) {
     // will call us again when it finishes.
     if (this.opening) return;
 
+    // This is responsible for closing any child popovers when a parent popover is closed
+    // it must run before the detach leaves logic below, otherwise the targetEls will no longer exist
```

Comment:
I think we can make this more robust by having updateLeaves() check if targetEl is still mounted -- i.e. if an unpinned popover has its targetEl disappear from the DOM, it can set onTarget false and transition().  That approach should handle the case where the target disappears due to leaf closure or navigation to a new file, as well as directly hiding the popover containing the target, and that approach could probably replace this chunk of code entirely.